src/otx/core/data/dataset/instance_segmentation.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXInstanceSegDataset."""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+import torch
+from datumaro import DatasetSubset, Image, Polygon
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.instance_segmentation import InstanceSegBatchDataEntity, InstanceSegDataEntity
+from otx.core.utils.mask_util import polygon_to_bitmap
+
+from .base import OTXDataset, Transforms
+
+
+class OTXInstanceSegDataset(OTXDataset[InstanceSegDataEntity]):
+    """OTXDataset class for instance segmentation."""
+
+    def __init__(self, dm_subset: DatasetSubset, transforms: Transforms) -> None:
+        super().__init__(dm_subset, transforms)
+        self.poly2mask = 'polygon' not in dm_subset.get_annotated_type()
```

Comment:
`self.poly2mask` serves the purpose of facilitating direct mask training. When processing item annotations, there are two possible scenarios: if a datumaro mask exists in annotated format, it will be utilized; otherwise, the polygon information will be employed for training.

In the provided code snippet:
```python
for annotation in item.annotations:
      if isinstance(annotation, Polygon):
          points = np.array(annotation.points).reshape(-1, 2).astype(np.int32)
          x1, y1 = np.min(points, axis=0)
          x2, y2 = np.max(points, axis=0)
          gt_bboxes = np.vstack((gt_bboxes, np.array([x1, y1, x2, y2])))
          gt_labels = np.append(gt_labels, annotation.label)
          if self.poly2mask:
              mask = polygon_to_bitmap([annotation], *img_shape)
              gt_masks = np.vstack((gt_masks, mask))
          else:
              gt_polygons.append(annotation)
```

Reply:
```python
self.poly2mask = 'polygon' not in dm_subset.get_annotated_type()
```
However, from this line, `self.poly2mask=True` means that there is no `datumaro.Polygon` in the subset.
So, it seems that it cannot reach to L47-48 because of L40.

https://github.com/openvinotoolkit/training_extensions/blob/e86f6e433a5e8cdb421234806ba02c10c7d964a1/src/otx/core/data/dataset/instance_segmentation.py#L39-L50

Is it correct?

Reply:
I agree. Currently, there isn't a well-defined method for passing this parameter through the data configuration. We have two potential approaches: either introduce a new parameter (`polygon2mask`) in `DataModuleConfig` or opt for a simpler solution by exclusively using `gt_polygons` and eliminating support for direct masks. Please let me know your thoughts on this.